### PR TITLE
Implemented generic napalm tool with debugging capabilities

### PR DESCRIPTION
**Pull request by dbarrosop**
Implemented a "show tech" type of command to gather enough data to start working with issues.

Note I added pre_connection, connection and post_connection method to the base driver so we can override on specifics drivers. This way we can run specific tests, enable debugging or show information for each particular platform. A `pre_connection_test` could easily be a telnet or curl to the device with the provided data and see if it works.

* example 1, driver failure
```
(napalm) rlyeh :: workspace/napalm/napalm-base ‹dbarrosop-show-tech› % cl_napalm_show_tech --user asd --password asdasd --vendor eosa 123
2017-08-03 11:35:26,676 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-03 11:35:26,676 - napalm - DEBUG - Gathering napalm packages
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-ansible==0.6.1
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-base==0.24.3
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-03 11:35:26,688 - napalm - DEBUG - napalm==1.2.0
2017-08-03 11:35:26,688 - napalm - DEBUG - get_network_driver - Calling with args: ('eosa',), {}
2017-08-03 11:35:26,688 - napalm - ERROR - get_network_driver - Failed: Cannot import "napalm_eosa". Is the library installed?

================= Traceback =================

Traceback (most recent call last):
  File "/Users/dbarroso/.virtualenvs/napalm/bin/cl_napalm_show_tech", line 11, in <module>
    load_entry_point('napalm-base', 'console_scripts', 'cl_napalm_show_tech')()
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 131, in main
    run_tests(args)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 108, in run_tests
    driver = test_get_network_driver(args.vendor)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 30, in wrapper
    r = func(*args, **kwargs)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 57, in test_get_network_driver
    return get_network_driver(vendor)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/__init__.py", line 103, in get_network_driver
    install_name=module_install_name
napalm_base.exceptions.ModuleImportError: Cannot import "napalm_eosa". Is the library installed?
```

* Example 2 - connection error

```
(napalm) rlyeh :: workspace/napalm/napalm-base ‹dbarrosop-show-tech› % cl_napalm_show_tech --user vagrant --password vagrant --vendor eos --optional_args "port=1244" localhost
2017-08-03 11:36:12,601 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-03 11:36:12,601 - napalm - DEBUG - Gathering napalm packages
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-ansible==0.6.1
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-base==0.24.3
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-03 11:36:12,614 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-03 11:36:12,615 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-03 11:36:12,615 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-03 11:36:12,615 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-03 11:36:12,615 - napalm - DEBUG - napalm==1.2.0
2017-08-03 11:36:12,615 - napalm - DEBUG - get_network_driver - Calling with args: ('eos',), {}
2017-08-03 11:36:12,631 - napalm - ERROR - get_network_driver - Successful
2017-08-03 11:36:12,631 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_eos.eos.EOSDriver'>, 'localhost', 'vagrant', 'vagrant', 60), {'optional_args': {u'port': u'1244'}}
2017-08-03 11:36:12,631 - napalm - ERROR - __init__ - Successful
2017-08-03 11:36:12,631 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10cf02850>,), {}
2017-08-03 11:36:12,631 - napalm - INFO - pre_connection_tests - Not implemented
2017-08-03 11:36:12,631 - napalm - DEBUG - open - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10cf02850>,), {}
No handlers could be found for logger "pyeapi.eapilib"
2017-08-03 11:36:12,633 - napalm - ERROR - open - Failed: Socket error during eAPI connection: [Errno 61] Connection refused

================= Traceback =================

Traceback (most recent call last):
  File "/Users/dbarroso/.virtualenvs/napalm/bin/cl_napalm_show_tech", line 11, in <module>
    load_entry_point('napalm-base', 'console_scripts', 'cl_napalm_show_tech')()
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 131, in main
    run_tests(args)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 114, in run_tests
    test_open_device(device)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 30, in wrapper
    r = func(*args, **kwargs)
  File "/Users/dbarroso/workspace/napalm/napalm-base/napalm_base/clitools/cl_napalm_show_tech.py", line 94, in test_open_device
    device.open()
  File "/Users/dbarroso/.virtualenvs/napalm/lib/python2.7/site-packages/napalm_eos/eos.py", line 118, in open
    raise ConnectionException(ce.message)
napalm_base.exceptions.ConnectionException: Socket error during eAPI connection: [Errno 61] Connection refused
```

* Example 3 -  all good

```
(napalm) rlyeh :: workspace/napalm/napalm-base ‹dbarrosop-show-tech› % cl_napalm_show_tech --user vagrant --password vagrant --vendor eos --optional_args "port=12443" localhost
2017-08-03 11:36:41,687 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-03 11:36:41,687 - napalm - DEBUG - Gathering napalm packages
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-ansible==0.6.1
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-base==0.24.3
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-03 11:36:41,699 - napalm - DEBUG - napalm==1.2.0
2017-08-03 11:36:41,699 - napalm - DEBUG - get_network_driver - Calling with args: ('eos',), {}
2017-08-03 11:36:41,708 - napalm - ERROR - get_network_driver - Successful
2017-08-03 11:36:41,708 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_eos.eos.EOSDriver'>, 'localhost', 'vagrant', 'vagrant', 60), {'optional_args': {u'port': u'12443'}}
2017-08-03 11:36:41,708 - napalm - ERROR - __init__ - Successful
2017-08-03 11:36:41,708 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,708 - napalm - INFO - pre_connection_tests - Not implemented
2017-08-03 11:36:41,708 - napalm - DEBUG - open - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,741 - napalm - ERROR - open - Successful
2017-08-03 11:36:41,741 - napalm - DEBUG - connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,741 - napalm - INFO - connection_tests - Not implemented
2017-08-03 11:36:41,742 - napalm - DEBUG - get_facts - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,768 - napalm - DEBUG - Gathered facts:
{u'fqdn': u'localhost',
 u'hostname': u'localhost',
 u'interface_list': [u'Ethernet1', u'Ethernet2', u'Management1'],
 u'model': u'vEOS',
 u'os_version': u'4.15.2.1F-2759627.41521F',
 u'serial_number': u'',
 u'uptime': 57151,
 u'vendor': u'Arista'}
2017-08-03 11:36:41,768 - napalm - ERROR - get_facts - Successful
2017-08-03 11:36:41,768 - napalm - DEBUG - close - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,769 - napalm - ERROR - close - Successful
2017-08-03 11:36:41,769 - napalm - DEBUG - post_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x10ff61850>,), {}
2017-08-03 11:36:41,769 - napalm - INFO - post_connection_tests - Not implemented
```

* Example 4 - users can even ask for a particular getter

```
(napalm) rlyeh :: workspace/napalm/napalm-base ‹dbarrosop-show-tech› % cl_napalm_show_tech --user vagrant --password vagrant --vendor eos --optional_args "port=12443" -g get_route_to -k destination=10.0.2.1 localhost
2017-08-03 11:37:11,711 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-03 11:37:11,711 - napalm - DEBUG - Gathering napalm packages
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-ansible==0.6.1
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-base==0.24.3
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-03 11:37:11,725 - napalm - DEBUG - napalm==1.2.0
2017-08-03 11:37:11,726 - napalm - DEBUG - get_network_driver - Calling with args: ('eos',), {}
2017-08-03 11:37:11,734 - napalm - ERROR - get_network_driver - Successful
2017-08-03 11:37:11,734 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_eos.eos.EOSDriver'>, 'localhost', 'vagrant', 'vagrant', 60), {'optional_args': {u'port': u'12443'}}
2017-08-03 11:37:11,734 - napalm - ERROR - __init__ - Successful
2017-08-03 11:37:11,734 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,734 - napalm - INFO - pre_connection_tests - Not implemented
2017-08-03 11:37:11,734 - napalm - DEBUG - open - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,768 - napalm - ERROR - open - Successful
2017-08-03 11:37:11,768 - napalm - DEBUG - connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,768 - napalm - INFO - connection_tests - Not implemented
2017-08-03 11:37:11,769 - napalm - DEBUG - get_facts - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,796 - napalm - DEBUG - Gathered facts:
{u'fqdn': u'localhost',
 u'hostname': u'localhost',
 u'interface_list': [u'Ethernet1', u'Ethernet2', u'Management1'],
 u'model': u'vEOS',
 u'os_version': u'4.15.2.1F-2759627.41521F',
 u'serial_number': u'',
 u'uptime': 57181,
 u'vendor': u'Arista'}
2017-08-03 11:37:11,797 - napalm - ERROR - get_facts - Successful
2017-08-03 11:37:11,797 - napalm - DEBUG - getter - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>, 'get_route_to'), {u'destination': u'10.0.2.1'}
2017-08-03 11:37:11,797 - napalm - DEBUG - get_route_to - Attempting to resolve getter
2017-08-03 11:37:11,797 - napalm - DEBUG - get_route_to - Attempting to call getter with kwargs: {u'destination': u'10.0.2.1'}
2017-08-03 11:37:11,830 - napalm - DEBUG - get_route_to - Response
{u'10.0.2.0/24': [{u'age': 0,
                   u'current_active': True,
                   u'inactive_reason': u'',
                   u'last_active': True,
                   u'next_hop': u'',
                   u'outgoing_interface': u'Management1',
                   u'preference': 0,
                   u'protocol': u'connected',
                   u'protocol_attributes': {},
                   u'routing_table': u'default',
                   u'selected_next_hop': True}]}
2017-08-03 11:37:11,831 - napalm - ERROR - getter - Successful
2017-08-03 11:37:11,831 - napalm - DEBUG - close - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,831 - napalm - ERROR - close - Successful
2017-08-03 11:37:11,831 - napalm - DEBUG - post_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x107251890>,), {}
2017-08-03 11:37:11,831 - napalm - INFO - post_connection_tests - Not implemented
```

### CODE REVIEW

**Comment by dbarrosop** [comment]
We could potentially extend with an `--attempt-configuration-change` kind of option.

**Comment by mirceaulinic** [comment]
Ready to merge @dbarrosop?

File: napalm_base/clitools/helpers.py [hunk 1]
```diff
@@ -88,6 +88,19 @@ def build_help(connect_test=False, validate=False, configure=False):
             action='store',
             help='Validation file containing resources derised states'
         )
+    elif show_tech:
+        parser.add_argument(
+            '--getter', '-g',
```
Comment (bewing) [inline napalm_base/clitools/helpers.py]: Maybe name this `method` instead?  My first try was `--getter facts`, which threw an error (expected `--getter get_facts`.  And `--getter ping --getter-kwargs 'destination=4.2.2.1` works, but looks weird

**Comment by dbarrosop** [comment]
Masking passwords shouldn't be hard. Will do.

File: napalm_base/clitools/helpers.py (hunk 1 above)
Comment (dbarrosop) [inline napalm_base/clitools/helpers.py]: Yeah, you are right. Will change.

**Comment by dbarrosop** [comment]
It also supports now doing configuration changes:

```
(napalm) ➜  napalm-base git:(dbarrosop-show-tech) ✗ cl_napalm_show_tech --user vagrant --password vagrant --vendor eos --optional_args "port=12443" --method get_interfaces -f new_config --strategy merge localhost
2017-08-14 19:56:11,359 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-14 19:56:11,360 - napalm - DEBUG - Gathering napalm packages
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-ansible==0.7.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-base==0.24.3
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-14 19:56:11,374 - napalm - DEBUG - napalm==1.2.0
2017-08-14 19:56:11,374 - napalm - DEBUG - get_network_driver - Calling with args: ('eos',), {}
2017-08-14 19:56:11,384 - napalm - ERROR - get_network_driver - Successful
2017-08-14 19:56:11,384 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_eos.eos.EOSDriver'>, 'localhost', 'vagrant'), {'password': u'*******', 'optional_args': {u'port': u'12443'}, 'timeout': 60}
2017-08-14 19:56:11,384 - napalm - ERROR - __init__ - Successful
2017-08-14 19:56:11,384 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:11,384 - napalm - INFO - pre_connection_tests - Not implemented
2017-08-14 19:56:11,384 - napalm - DEBUG - open - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:11,418 - napalm - ERROR - open - Successful
2017-08-14 19:56:11,418 - napalm - DEBUG - connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:11,418 - napalm - INFO - connection_tests - Not implemented
2017-08-14 19:56:11,418 - napalm - DEBUG - get_facts - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:11,444 - napalm - DEBUG - Gathered facts:
{u'fqdn': u'asdasd',
 u'hostname': u'asdasd',
 u'interface_list': [u'Ethernet1', u'Ethernet2', u'Management1'],
 u'model': u'vEOS',
 u'os_version': u'4.15.2.1F-2759627.41521F',
 u'serial_number': u'',
 u'uptime': 2144,
 u'vendor': u'Arista'}
2017-08-14 19:56:11,444 - napalm - ERROR - get_facts - Successful
2017-08-14 19:56:11,444 - napalm - DEBUG - getter - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>, 'get_interfaces'), {}
2017-08-14 19:56:11,444 - napalm - DEBUG - get_interfaces - Attempting to resolve getter
2017-08-14 19:56:11,444 - napalm - DEBUG - get_interfaces - Attempting to call getter with kwargs: {}
2017-08-14 19:56:11,475 - napalm - DEBUG - get_interfaces - Response
{u'Ethernet1': {u'description': u'',
                u'is_enabled': True,
                u'is_up': True,
                u'last_flapped': 1502731278.4342597,
                u'mac_address': u'08:00:27:E6:4C:E9',
                u'speed': 0},
 u'Ethernet2': {u'description': u'',
                u'is_enabled': True,
                u'is_up': True,
                u'last_flapped': 1502731278.4344258,
                u'mac_address': u'08:00:27:3D:83:34',
                u'speed': 0},
 u'Management1': {u'description': u'',
                  u'is_enabled': True,
                  u'is_up': True,
                  u'last_flapped': 1502731294.5988464,
                  u'mac_address': u'08:00:27:7D:44:C1',
                  u'speed': 1000}}
2017-08-14 19:56:11,475 - napalm - ERROR - getter - Successful
2017-08-14 19:56:11,475 - napalm - DEBUG - load_merge_candidate - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {'filename': 'new_config'}
2017-08-14 19:56:11,753 - napalm - ERROR - load_merge_candidate - Successful
2017-08-14 19:56:11,753 - napalm - DEBUG - compare_config - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
@@ -8,7 +8,7 @@
 !
 transceiver qsfp default-mode 4x10G
 !
-hostname asdasd
+hostname myhost
 !
 spanning-tree mode mstp
 !
2017-08-14 19:56:12,562 - napalm - ERROR - compare_config - Successful
2017-08-14 19:56:12,562 - napalm - DEBUG - commit_config - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:13,153 - napalm - ERROR - commit_config - Successful
2017-08-14 19:56:13,153 - napalm - DEBUG - close - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:13,153 - napalm - ERROR - close - Successful
2017-08-14 19:56:13,153 - napalm - DEBUG - post_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11156ad50>,), {}
2017-08-14 19:56:13,153 - napalm - INFO - post_connection_tests - Not implemented
```

**Comment by dbarrosop** [comment]
Ok, so with the last commit I went overboard and turned this into a generic method that can call getters and do configuration changes. It has a regular and a debug mode. For instance:

```
(napalm) ➜  napalm-base git:(dbarrosop-show-tech) ✗ napalm --user vagrant --password vagrant --vendor eos --optional_args "port=12443" --method get_interfaces -f new_config --strategy merge --dry-run localhost
2017-08-14 20:10:55,860 - napalm - INFO - Result of get_interfaces:
{
    "Ethernet2": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731278.4344149,
        "is_up": true,
        "mac_address": "08:00:27:3D:83:34",
        "speed": 0
    },
    "Management1": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731294.5988362,
        "is_up": true,
        "mac_address": "08:00:27:7D:44:C1",
        "speed": 1000
    },
    "Ethernet1": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731278.4342613,
        "is_up": true,
        "mac_address": "08:00:27:E6:4C:E9",
        "speed": 0
    }
}
2017-08-14 20:10:57,237 - napalm - INFO - Gathered diff:
@@ -8,7 +8,7 @@
 !
 transceiver qsfp default-mode 4x10G
 !
-hostname myhost
+hostname myhost2
 !
 spanning-tree mode mstp
 !
(napalm) ➜  napalm-base git:(dbarrosop-show-tech) ✗ napalm --user vagrant --password vagrant --vendor eos --optional_args "port=12443" --method get_interfaces -f new_config --strategy merge --dry-run --debug localhost
2017-08-14 20:11:07,262 - napalm - DEBUG - Starting napalm's debugging tool
2017-08-14 20:11:07,262 - napalm - DEBUG - Gathering napalm packages
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-ansible==0.7.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-base==0.24.3
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-eos==0.6.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-fortios==0.4.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-ios==0.7.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-iosxr==0.5.4
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-junos==0.12.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-nxos==0.6.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-panos==0.4.0
2017-08-14 20:11:07,274 - napalm - DEBUG - napalm-pluribus==0.5.1
2017-08-14 20:11:07,275 - napalm - DEBUG - napalm-ros==0.2.2
2017-08-14 20:11:07,275 - napalm - DEBUG - napalm-vyos==0.1.3
2017-08-14 20:11:07,275 - napalm - DEBUG - napalm==1.2.0
2017-08-14 20:11:07,275 - napalm - DEBUG - get_network_driver - Calling with args: ('eos',), {}
2017-08-14 20:11:07,286 - napalm - DEBUG - get_network_driver - Successful
2017-08-14 20:11:07,286 - napalm - DEBUG - __init__ - Calling with args: (<class 'napalm_eos.eos.EOSDriver'>, 'localhost', 'vagrant'), {'password': u'*******', 'optional_args': {u'port': u'12443'}, 'timeout': 60}
2017-08-14 20:11:07,286 - napalm - DEBUG - __init__ - Successful
2017-08-14 20:11:07,286 - napalm - DEBUG - pre_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:07,286 - napalm - DEBUG - open - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:07,354 - napalm - DEBUG - open - Successful
2017-08-14 20:11:07,354 - napalm - DEBUG - connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:07,354 - napalm - DEBUG - get_facts - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:07,412 - napalm - DEBUG - Gathered facts:
{
    "os_version": "4.15.2.1F-2759627.41521F",
    "uptime": 3040,
    "interface_list": [
        "Ethernet1",
        "Ethernet2",
        "Management1"
    ],
    "vendor": "Arista",
    "serial_number": "",
    "model": "vEOS",
    "hostname": "myhost",
    "fqdn": "myhost"
}
2017-08-14 20:11:07,412 - napalm - DEBUG - get_facts - Successful
2017-08-14 20:11:07,412 - napalm - DEBUG - getter - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>, 'get_interfaces'), {}
2017-08-14 20:11:07,412 - napalm - DEBUG - get_interfaces - Attempting to resolve getter
2017-08-14 20:11:07,412 - napalm - DEBUG - get_interfaces - Attempting to call getter with kwargs: {}
2017-08-14 20:11:07,465 - napalm - DEBUG - get_interfaces - Response
2017-08-14 20:11:07,465 - napalm - INFO - Result of get_interfaces:
{
    "Ethernet2": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731278.4344132,
        "is_up": true,
        "mac_address": "08:00:27:3D:83:34",
        "speed": 0
    },
    "Management1": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731294.5988345,
        "is_up": true,
        "mac_address": "08:00:27:7D:44:C1",
        "speed": 1000
    },
    "Ethernet1": {
        "is_enabled": true,
        "description": "",
        "last_flapped": 1502731278.4342616,
        "is_up": true,
        "mac_address": "08:00:27:E6:4C:E9",
        "speed": 0
    }
}
2017-08-14 20:11:07,465 - napalm - DEBUG - getter - Successful
2017-08-14 20:11:07,465 - napalm - DEBUG - load_merge_candidate - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {'filename': 'new_config'}
2017-08-14 20:11:07,941 - napalm - DEBUG - load_merge_candidate - Successful
2017-08-14 20:11:07,941 - napalm - DEBUG - compare_config - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:08,681 - napalm - INFO - Gathered diff:
@@ -8,7 +8,7 @@
 !
 transceiver qsfp default-mode 4x10G
 !
-hostname myhost
+hostname myhost2
 !
 spanning-tree mode mstp
 !
2017-08-14 20:11:08,681 - napalm - DEBUG - compare_config - Successful
2017-08-14 20:11:08,681 - napalm - DEBUG - close - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
2017-08-14 20:11:08,723 - napalm - DEBUG - close - Successful
2017-08-14 20:11:08,723 - napalm - DEBUG - post_connection_tests - Calling with args: (<napalm_eos.eos.EOSDriver object at 0x11073ad90>,), {}
```

I figured we can obsolete all the methods we have now and just have a single ``napalm`` one?

**Comment by dbarrosop** [comment]
This one should be ready. I added deprecation warnings to the old cli tools. Would appreciate some testing :P

**Pull request merged**